lib/resourcemerge/core.go

```diff
@@ -466,16 +515,17 @@ func ensureVolumeSourceDefaults(required *corev1.VolumeSource) {
 }
 
 func ensureSecurityContextPtr(modified *bool, existing **corev1.SecurityContext, required *corev1.SecurityContext) {
-	// if we have no required, then we don't care what someone else has set
```

Comment:
This is originally from d9f6718de (#7), but there is no motivation there for why we used this permissive approach for this property.

Reply:
ACK. I commented same in Slack. Was hoping you may have some history or knowledge of why it was done this way but guess not :)

Reply:
and now documented in the commit message :bow: